### PR TITLE
1219 - Quest - Patient attributes and Address attributes

### LINKED ISSUE #1219: Add and extract attributes to Patient model
Currently, in the Quest app, there is a Patient model that only has the following attributes: id, identifier, name, gender, age, address, & additional data. However, for the Tingathe app, we need some additional attributes that we are saving to the server.

When looking at the Patient FHIR specification, we need the following additional attributes:

- telcom
- address
- contact
- generalPractitioner
- managingOrganization
- link

The Patient model does have an address attribute but it is extracted as a string as a result of a combination of the individual address attributes. However, we need to be able to access the individual attributes within the address itself. Specifically, we need to access:

- district
- state
- text

### PR DESCRIPTION
**Pull request by AbdulWahabMemon**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1219 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by KhumboLihonga** [comment]
This looks great. Just a quick comment, telecom should be a List of strings as a CHW should be able to save multiple phone numbers for a patient.

**Comment by ellykits** [comment]
Can we discuss how to include these attributes to the model that will be used in the single apk architecture?

**Comment by AbdulWahabMemon** [comment]
> @AbdulWahabMemon This looks great. Just a quick comment, telecom should be a List of strings as a CHW should be able to save multiple phone numbers for a patient.

@KhumboLihonga patient model attribute "telecom" also updated with list of strings

**Comment by AbdulWahabMemon** [comment]
@dubdabasoduba for SAA refactoring and targeting with implement-generic-app-register, there is seperate PR #1239.

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt [hunk 1]
```diff
@@ -135,6 +135,36 @@ fun Patient.extractAddress(): String {
   }
 }
 
+fun Patient.extractAddressDistrict(): String {
+  if (!hasAddress()) return ""
+  return with(addressFirstRep) { this.district ?: "" }
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt:140]: can we combine these 2 lines w/a ternary here and below?

```kotlin
return with(addressFirstRep) { this.district ?: "" } if (hasAddress()) else ""
```

**Review by ellykits — APPROVED** [review]

**Pull request merged**